.github/workflows/deployment.yaml

```diff
@@ -277,3 +277,18 @@ jobs:
           app-name: "$TF_ENV-read-source-data-$SHORT_CID"
           package: serverless-functions
           publish-profile: ${{ env.PUBLISH_PROFILE }}
+
+      - name: Add Event Grid subscription
+        env:
+          SUBSCRIPTION_ID: ${{ secrets.SUBSCRIPTION_ID }}
+          RESOURCE_GROUP_NAME: ${{ secrets.RESOURCE_GROUP_NAME }}
+          TF_ENV: ${{ steps.set-environment.outputs.tf_env }}
+          SHORT_CID: ${{ steps.set-environment.outputs.short_cid }}
+        run: |
+          az eventgrid event-subscription create \
+            --name phdi${TF_ENV}phi${SHORT_CID}Sub \
+            --source-resource-id "/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP_NAME/providers/Microsoft.Storage/storageaccounts/phdi${TF_ENV}phi${SHORT_CID}" \
+            --included-event-types Microsoft.Storage.BlobCreated \
+            --subject-begins-with /blobServices/default/containers/source-data/blobs/ \
+            --endpoint "/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP_NAME}/providers/Microsoft.Web/sites/${TF_ENV}-read-source-data-${SHORT_CID}/functions/ReadSourceData" \
+            --endpoint-type azurefunction
```

Comment:
Why can't this just be done with Terraform?

Reply:
Because the `ReadSourceData` function has to exist before the EventGrid subscription can be created, but we don't create the "function" with TF, we only create the  "function app" and then deploy the "function" using the GH action, so this has to be done after that. Sorry, I meant to mention this in the PR description but forgot!

Reply:
Hmmm okay. I remember the trouble with deploying the "function" via TF. I guess this works for now.